spring-cloud/spring-cloud-stream/spring-cloud-stream-kafka/src/main/java/com/baeldung/config/SchemRegistryConfig.java

```diff
@@ -0,0 +1,21 @@
+package com.baeldung.config;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.stream.schema.client.ConfluentSchemaRegistryClient;
+import org.springframework.cloud.stream.schema.client.SchemaRegistryClient;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class SchemRegistryConfig {
+
+    @Value("${spring.cloud.stream.kafka.binder.producer-properties.schema.registry.url}")
+    private String endPoint;
+
+    @Bean
+    public SchemaRegistryClient schemaRegistryClient() {

```

Comment:
Consider just injecting the `@Value(...)` into this method, rather than using field injection to store it.

Reply:
done